src/components/Card.js

```diff
@@ -27,10 +49,19 @@ export class Card {
   _setEventListeners() {
     this._templateClone
       .querySelector(".card__button")
-      .addEventListener("click", this._likeCard);
+      .addEventListener("click", (event) => {
+        this.handleLike(this.id, this.isLiked)
+          .then((res) => res.json())
+          .then((data) => console.log(data));
+        this._likeCard(event);
+        this.isLiked = !this.isLiked;
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.